autopush/websocket.py

```diff
@@ -366,14 +369,27 @@ def _check_other_nodes(self, result):
             node_id = self._get_aval(attrs.get("node_id", {}), 'S')
             uaid = self._get_aval(attrs.get("uaid", {}), 'S')
             connected_at = self._get_aval(attrs.get("connected_at", {}), 'N')
-            url = "%s/notif/%s/%s" % (node_id, uaid, connected_at)
-            if not getattr(self, 'testing', False):
-                d = self.ap_settings.agent.request(
-                    "DELETE",
-                    url.encode("utf8"),
-                ).addCallback(self.finish_hello)
-                d.addErrback(self.log_err, extra="Failed to delete old node")
-                return
+            inSelf = self.ap_settings.clients.get(uaid)
+            if inSelf:
+                if inSelf.connected_at <= connected_at:
+                    inSelf.sendClose()
+            else:
+                url = "%s/notif/%s/%s" % (node_id, uaid, connected_at)
+
+                def _eatConnections(*args):
+                    failure.trap(
+                        ConnectError, ConnectionRefusedError, UserError
+                    )
+
+                if not getattr(self, 'testing', False):
+                    self.ap_settings.agent.request(
+                        "DELETE",
+                        url.encode("utf8"),
+                    ).addErrback(
+                        _eatConnections
```

Comment:
And not to nit too much, but since we're back in Python-land, while twisted unfortunately uses camel-case excessively, our own code should generally try to stick with underscore separated lower-case.